pkg/kube/cluster-init.sh

```diff
@@ -7,59 +7,96 @@ K3S_VERSION=v1.26.3+k3s1
 KUBEVIRT_VERSION=v0.59.0
 LONGHORN_VERSION=v1.4.2
 CDI_VERSION=v1.56.0
+NODE_IP=""
 
 INSTALL_LOG=/var/lib/install.log
 CTRD_LOG=/var/lib/containerd.log
 LOG_SIZE=$((5*1024*1024))
 
 logmsg() {
-   local MSG
-   local TIME
-   MSG="$*"
-   TIME=$(date +"%F %T")
-   echo "$TIME : $MSG"  >> $INSTALL_LOG
+        local MSG
+        local TIME
+        MSG="$*"
+        TIME=$(date +"%F %T")
+        echo "$TIME : $MSG"  >> $INSTALL_LOG
 }
 
 setup_cgroup () {
-   echo "cgroup /sys/fs/cgroup cgroup defaults 0 0" >> /etc/fstab
+        echo "cgroup /sys/fs/cgroup cgroup defaults 0 0" >> /etc/fstab
 }
 
 
 check_network_connection () {
- while true; do
-
-   ret=$(curl -o /dev/null -w "%{http_code}" -s "https://get.k3s.io")
-   if [ "$ret" -eq 200 ]; then
-        logmsg "Network is ready."
-        break;
-   else
-        logmsg "Network is not yet ready"
-   fi
-
-   sleep 5
- done
+        while true; do
+                ret=$(curl -o /dev/null -w "%{http_code}" -s "https://get.k3s.io")
+                if [ "$ret" -eq 200 ]; then
+                        logmsg "Network is ready."
+                        break;
+                else
+                        logmsg "Network is not yet ready"
+                fi
+                sleep 5
+        done
 }
 
 wait_for_default_route() {
-  while read -r iface dest gw flags refcnt use metric mask mtu window irtt; do
-    if [ "$dest" = "00000000" ] && [ "$mask" = "00000000" ]; then
-      logmsg "Default route found"
-      return 0
-    fi
-    #Make yetus happy
-    logmsg "waiting for default route $iface $dest $gw $flags $refcnt $use $metric $mask $mtu $window $irtt"
-    sleep 1
-  done < /proc/net/route
-
-  return 1
+        while read -r iface dest gw flags refcnt use metric mask mtu window irtt; do
+                if [ "$dest" = "00000000" ] && [ "$mask" = "00000000" ]; then
+                        logmsg "Default route found"
+                        return 0
+                fi
+                logmsg "waiting for default route $iface $dest $gw $flags $refcnt $use $metric $mask $mtu $window $irtt"
+                sleep 1
+        done < /proc/net/route
+        return 1
+}
+
+# Get IP of the interface with the first default route
+# This will be then used as K3s node IP.
```

Comment:
How does this relate to the cluster interface which the controller will provide via a new API? Are they unrelated? Or this is an initial implementation which works without any API additions?

Reply:
Yes, this is just an initial implementation until we can specify interface over some API addition.